### PR TITLE
Add support for `rustdoc` shorthand intra-doc links

### PR DESCRIPTION
**Pull request by virtualritz**
## What

`rustdoc` lets you link to an item by writing its path in a code span -- e.g. ``[`my_function`]`` -- with no explicit destination. `cargo-rdme` left these untouched, so the generated README ended up with a broken ``[`my_function`]`` instead of a link.

This teaches `cargo-rdme` to recognize and process the shorthand ``[`item`]`` form:

- Resolvable shorthands become docs.rs links, e.g. ``[`my_function`](https://docs.rs/<crate>/latest/<crate>/fn.my_function.html)``.
- Unresolvable shorthands (with the usual warning) are stripped to a plain code span, instead of leaving broken link syntax in the output.
- In `--strip-links` mode they are reduced to a plain code span.

## Does not disturb existing links

Detection deliberately leaves alone the markdown link and reference constructs a user wrote on purpose, so it never corrupts them:

- inline links ``[`x`](url)``,
- reference links ``[`x`][label]``,
- reference definitions and shortcut references ``[`x`]: ...``.

Markdown requires `(`/`[` to *immediately* follow the closing `]`; a space (e.g. ``[`f()`] (fast)``) means it is not a link, so such shorthands are still converted.

## Tests

- `transform_intralinks_shorthand` -- conversion, stripping, and edge cases (lists, multiple per line, already-converted, spaced paren).
- `transform_intralinks_shorthand_refs` -- asserts reference links and definitions are preserved byte-for-byte alongside bare-shorthand conversion.

### CODE REVIEW

File: tests/transform_intralinks_shorthand/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[package]
+name = "integration_test"
+version = "0.0.0"
+edition = "2021"
```
Comment (orium) [inline tests/transform_intralinks_shorthand/Cargo.toml:4]: use edition 2024 in this and the other integration test.

**Comment by orium** [comment]
can you rebase please?

**Comment by virtualritz** [comment]
Rebased onto latest `main`.

**Comment by orium** [comment]
I've checked the code and there are a few things I don't like. For instance, parsing markdown by hand is a big no-no for me. This is something I'll to implement myself to make sure I'm happy with the result. I'll only do that after #236 lands.
